sound/soc/sof/pm.c

```diff
@@ -422,3 +438,41 @@ int snd_sof_suspend(struct device *dev)
 	return sof_suspend(dev, false);
 }
 EXPORT_SYMBOL(snd_sof_suspend);
+
+int snd_sof_set_dsp_power_state(struct snd_sof_dev *sdev,
+				enum sof_d0_substate d0_substate)
+{
+	u32 flags;
+	int ret;
+
+	dev_dbg(sdev->dev, "setting ADSP D0 substate:%d\n", d0_substate);
+
+	/* Todo: destroy/reload unused pipelines/widgets in FW */
+
+	/* do platform specific set_state */
+	ret = snd_sof_dsp_set_power_state(sdev, d0_substate);
```

Comment:
snd_sof_set_dsp_power_state() calls snd_sof_dsp_set_power_state() ?

NAK right there.

Reply:
um, how about changing snd_sof_set_dsp_power_state() to snd_sof_set_d0_state()?